File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* removeNthFromEnd(struct ListNode* head, int n) {
    struct ListNode *temp1=head,*temp2=head,*len=head;
    int j=0;
    while(len!=NULL){
        j++;
        len=len->next;
    }
    if(j==1){
        head=head->next;
    }
    else{
        while(j!=n){
             temp2=temp1;
        temp1=temp1->next;
         j--;
           }
        if(temp1!=temp2){
        temp2->next=temp1->next;}
        else{
            head=head->next;
        }
        
    }
    return head;
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode temp=head,temp1=head;
        int l=0,i=0;
        while(temp!=null){l++;
        temp=temp.next;}
        int d=l-n;
        if(d==0){return head.next;}
        for(i=0;i<d-1;i++){
            temp1=temp1.next;
        }
        temp1.next=temp1.next.next;
        return head;
    }

}